### ISSUE TITLE
Event replace_near_blocks radius just looks in positive direction

### ISSUE THREAD

**Issue by tinkeringmatt** [PluginBugs/Issues-ItemsAdder#1668]
### Terms

- [X] I'm using the very latest version of ItemsAdder and its dependencies.
- [X] I already searched on this [Github page](https://github.com/PluginBugs/Issues-ItemsAdder/issues) to check if the same issue was already reported.
- [X] I already searched on the [plugin wiki](https://itemsadder.devs.beer/) to know if a solution is already known.
- [X] I already asked on the **#💬ia-community-help** channel on **Discord** to know if anyone already has a solution for the issue.

### Discord tag (optional)

Tink (he/they/she)#3461

### What happened?

Event in question

events:
      block_break:
        replace_near_blocks:
          radius:
            x: 3
            y: 3
            z: 3
          from: STONE
          to: SAND

while using this, it essentially doesn't do a "radius" around the broken block, instead it just does a search that only goes in the positive directions in all three axis

So, instead of checking the positions around the broken block, the positions checked treat the broken block like its at the bottom corner of the area.

Let me know if you need any more detail, as this bug has existed for as long as Ive tried using it 

https://user-images.githubusercontent.com/18208248/165907196-af5f4baa-49a6-499e-ba5b-191de8c54b95.mp4

### Steps to reproduce the issue

Use replace_near_blocks with a larger "radius"
Break a block
look at what blocks where changed and what blocks are uneffected

### Server version

Paper-312 
1.18.2

### ItemsAdder Version

3.1.2

### ProtocolLib Version

4.8.0

### LoneLibs Version

1.0.18

### LibsDisguises Version (optional)

10.0.28

### ItemsAdder config.yml

```shell
metrics: true
resource-pack:
  apply-on-join: true
  kick-player-on-decline: false
  kick-player-on-fail: false
  delay-ticks: 1
  custom-font:
    enabled: true
    name: ipaexg.ttf
    shift: '[0,0]'
    size: 11.0
    oversample: 2.0
  thin-font:
    enabled: false
  zip:
    ignore-files-ext:
    - db
    - yml
    - xcf
    - html
    fix_transparent_noteblocks__disable-for-more-performance: false
    protect-file-from-unzip:
      enabled: true
      extreme: true
    overwrite-vanilla-models: true
    overwrite-vanilla-lang-json: true
    compress-png-files: true
    compress-json-files: true
    generate-custom-armors-textures:
      vanilla_1_17: true
      optifine: false
    disable-overwrite-leather-armor-layers-textures: false
  title:
    enabled: false
  notify-other-plugins-interfering: false
  protect-player:
    black-screen: false
    hide-hud: false
    cancel-movement: true
    lock-player: true
  hosting:
    no-host:
      enabled: false
    auto-external-host:
      enabled: false
    self-host:
      enabled: true
      server-ip: 77.75.125.178
      pack-port: 8123
    external-host:
      enabled: false
      url: ''
      skip-url-file-type-check___DONT_ASK_HELP_IF_SET_TRUE: false
  items:
    CustomModelData-starting-value:
      ALL: 10000
      DIAMOND: 11000
  command:
    usage-cooldown-seconds: 60
blocks:
  light-api: true
  cancel-break-wrong-tool:
    enabled: false
    send-actionbar: true
  campfire:
    unlit-on-place: false
    allow-any-item: true
  fix-glitched-blocks:
    enabled: true
    only-new-chunks: false
    fix-real-blocks-drops-chorus-water-bug__disable-for-more-performance: false
    seach-also-in-these-biomes: {}
    search-also-in-these-biomes:
      enabled: false
      biomes:
      - BEACH
      - BIRCH_FOREST
      - BIRCH_FOREST_HILLS
      - FOREST
      - MOUNTAINS
      - MOUNTAIN_EDGE
      - GRAVELLY_MOUNTAINS
      - PLAINS
      - RIVER
  disable-REAL: false
  disable-REAL_TRANSPARENT: false
  disable-REAL_NOTE: false
  disable-REAL_WIRE: false
  disable-FIRE: false
  fix-liquids-flow: true
  fix-floating-blocks: true
items:
  fix-show-attack-and-damage-as-lore: true
  give_death_location_maps: false
  hide-custom-durability-lore: true
font_images:
  chat:
    enabled: true
    replace-only-packets: true
    doublecheck-permission__disable-for-more-performance: false
    add-hover-suggestion: true
  command:
    enabled: true
    commandblocks: true
    excluded:
    - home
    - sethome
    - tpa
  sign:
    enabled: true
  book:
    enabled: true
  anvil:
    enabled: true
  inventory-title:
    enabled: true
  tab-header-footer:
    enabled: true
  vault-prefix-suffix:
    enabled: true
    update-every-ticks: 1200
  scoreboard-teams:
    enabled: true
    update-every-ticks: 3600
  player-display-name:
    enabled: true
    update-every-ticks: 1200
  replace-in-customitems-name-and-lore: true
huds:
  enabled: true
config_files:
  update-automatically: false
  lang: en
  extract-default-items: false
  extract-default-resources: false
  dictionaries-lang: en
debug:
  items:
    log-files-loading: false
    log-items-loading: false
    log-items-loading-disabled: false
    log-font-images-loading: false
    log-huds-loading: false
    log-attribute-modifiers-loading: false
    log-warnings: false
    log-loots-loading: false
    log-loots-loading-disabled: false
    log-armors-loading: false
  recipes:
    log-recipes-loading: false
    log-recipes-loading-disabled: false
    log-recipes-removed-vanilla: false
    log-fix-packet-too-large: false
  resource-pack:
    log-extract-default-resources: false
    log-not-extract-default-resources-already-exist: false
  blocks:
    log-save: false
    log-save-on-unload: false
    log-mined-unknown-blocks: true
  worlds-populators:
    log-loading: false
    log-loading-disabled: false
    log-loading-no-populators: false
    log-generation: false
    log-loading-no-world: true
  player-stats:
    log-save: true
  performance:
    loading-ms: true
  huds:
    log-triggers: false
  other:
    log-dictionaries-loading: false
    log-liquids-loading: false
    log-minecraft_lang_overwrite-loading: false
  categories:
    log-total-loaded: true
    log-single-loaded: false
    log-single-loaded-overwrite: false
    log-single-skip: false
    log-icon-failed-load: false
  trees-populators:
    log-loading: false
    log-loading-disabled: false
    log-loading-no-populators: false
    log-generation: false
    log-loading-no-world: true
  surface-decorators:
    log-loading: false
    log-loading-disabled: false
    log-loading-no-decorators: false
    log-generation: false
    log-loading-no-world: true
  entities:
    log-files-loading: false
    log-entities-loading: false
    log-entities-loading-disabled: false
liquids:
  enabled: true
disable-plugin-loading-messages_DONT_ASK_ME_FOR_SUPPORT_IF_YOU_SET_TRUE__THANKS: false
mobs:
  disable-MOB-behaviour__DISABLE_AT_YOUR_OWN_RISK: false
update-checker: true
entities:
  max-furniture-vehicles-per-chunk: 30
  allow-removing-old-furnitures_2_3_11: false
  custom-entitites:
    enabled: true
    interval-rendering-packets-ticks: 1
    emotes: true
recipes:
  remove-vanilla-recipes:
    enabled: false
    list:
      DIAMOND_SWORD: false
      IRON_INGOT: false
  hide-vanilla-unrecognized-recipe-errors: true
  custom-recipes:
    unlock:
      all-on-join: true
      on-pickup:
        any: false
        ingredients: true
  fix-packet-too-large-kick: false
effects:
  hide-scoreboard-numbers: false
  text-effects:
    enabled: true
    replace-in-customitems-name-and-lore: true
    chat:
      enabled: true
    sign:
      enabled: true
    book:
      enabled: true
    anvil:
      enabled: true
    tab-header-footer:
      enabled: true
loots:
  allow-loots-drop-from-spawners-using-silk-touch: false
```

### Problematic items yml configuration file (optional)

```shell
info:
  namespace: itemsadder
items:
  ruby_axe:
    display_name: Ruby Axe
    resource:
      material: DIAMOND_AXE
      generate: true
      model_id: 1
      textures:
      - item/ruby_axe.png
    attribute_modifiers:
      mainhand:
        attackDamage: 9.0
        attackSpeed: 1.0
      offhand:
        attackSpeed: 0.04
  spinel_axe:
    display_name: Spinel Axe
    resource:
      material: DIAMOND_AXE
      generate: true
      model_id: 2
      textures:
      - item/spinel_axe.png
    durability:
      max_custom_durability: 1162
    attribute_modifiers:
      mainhand:
        maxHealth: 4.0
        attackDamage: 9.0
        attackSpeed: 0.9
      offhand:
        maxHealth: 4.0
  amethyst_axe:
    display_name: Amethyst Axe
    resource:
      material: DIAMOND_AXE
      generate: true
      model_id: 5
      textures:
      - item/amethyst_axe.png
  stone_hammer:
    display_name: '&3&lStone hammer'
    resource:
      material: STONE_PICKAXE
      generate: true
      model_id: 1
      textures:
      - item/stone_hammer.png
    durability:
      max_custom_durability: 100
    events:
      block_break:
        replace_near_blocks:
          radius:
            x: 3
            y: 3
            z: 3
          from: STONE
          to: SAND
    attribute_modifiers:
      mainhand:
        attackDamage: 12.0
        attackSpeed: 0.4
  diamond_hammer:
    display_name: '&3&lDiamond hammer'
    resource:
      material: DIAMOND_PICKAXE
      generate: true
      model_id: 2
      textures:
      - item/diamond_hammer.png
    events:
      block_break:
        multiple_break:
          keep_ores: true
          drop_all_blocks:
            enabled: false
            need_silk_touch: true
          size: 3
    attribute_modifiers:
      mainhand:
        attackDamage: 14.0
        attackSpeed: 0.5
  demoniac_pickaxe:
    display_name: '&4&lDemoniac Pickaxe'
    resource:
      material: DIAMOND_PICKAXE
      generate: true
      model_id: 3
      textures:
      - item/demoniac_pickaxe.png
  ruby_pickaxe:
    display_name: Ruby Pickaxe
    resource:
      material: DIAMOND_PICKAXE
      generate: true
      model_id: 4
      textures:
      - item/ruby_pickaxe.png
    attribute_modifiers:
      mainhand:
        attackDamage: 4.0
        attackSpeed: 1.3
      offhand:
        attackSpeed: 0.06
  spinel_pickaxe:
    display_name: Spinel Pickaxe
    resource:
      material: DIAMOND_PICKAXE
      generate: true
      model_id: 5
      textures:
      - item/spinel_pickaxe.png
    durability:
      max_custom_durability: 1162
    attribute_modifiers:
      mainhand:
        maxHealth: 6.0
        attackDamage: 4.0
        attackSpeed: 1.2
      offhand:
        maxHealth: 6.0
  amethyst_pickaxe:
    display_name: Amethyst Pickaxe
    resource:
      material: DIAMOND_PICKAXE
      generate: true
      model_id: 8
      textures:
      - item/amethyst_pickaxe.png
    attribute_modifiers:
      mainhand:
        attackDamage: 4.0
        attackSpeed: 1.2
  ruby_hoe:
    display_name: Ruby hoe
    resource:
      material: DIAMOND_HOE
      generate: true
      model_id: 1
      textures:
      - item/ruby_hoe.png
    attribute_modifiers:
      mainhand:
        attackDamage: 1.0
        attackSpeed: 3.3
      offhand:
        attackSpeed: 0.08
  spinel_hoe:
    display_name: Spinel hoe
    resource:
      material: DIAMOND_HOE
      generate: true
      model_id: 2
      textures:
      - item/spinel_hoe.png
    durability:
      max_custom_durability: 1162
    attribute_modifiers:
      mainhand:
        maxHealth: 8.0
        attackDamage: 1.0
        attackSpeed: 3.0
      offhand:
        maxHealth: 8.0
  amethyst_hoe:
    display_name: Amethyst hoe
    resource:
      material: DIAMOND_HOE
      generate: true
      model_id: 5
      textures:
      - item/amethyst_hoe.png
    attribute_modifiers:
      mainhand:
        attackDamage: 1.0
        attackSpeed: 3.0
  ruby_shovel:
    display_name: Ruby Shovel
    resource:
      material: DIAMOND_SHOVEL
      generate: true
      model_id: 1
      textures:
      - item/ruby_shovel.png
  spinel_shovel:
    display_name: Spinel Shovel
    resource:
      material: DIAMOND_SHOVEL
      generate: true
      model_id: 2
      textures:
      - item/spinel_shovel.png
    durability:
      max_custom_durability: 1162
    attribute_modifiers:
      mainhand:
        maxHealth: 6.0
        attackDamage: 4.5
        attackSpeed: 1.0
      offhand:
        maxHealth: 6.0
  amethyst_shovel:
    display_name: Amethyst Shovel
    resource:
      material: DIAMOND_SHOVEL
      generate: true
      model_id: 5
      textures:
      - item/amethyst_shovel.png
  demoniac_hammer:
    display_name: '&4&lDemoniac Hammer'
    resource:
      material: DIAMOND_PICKAXE
      generate: true
      model_id: 10
      textures:
      - item/demoniac_hammer.png
    events:
      block_break:
        multiple_break:
          keep_ores: true
          drop_all_blocks:
            enabled: false
            need_silk_touch: true
          size: 3
          depth: 3
  sapphire_axe:
    display_name: Sapphire Axe
    resource:
      material: DIAMOND_AXE
      generate: true
      model_id: 7
      textures:
      - item/sapphire_axe.png
    attribute_modifiers:
      mainhand:
        attackDamage: 8.0
        attackSpeed: 1.0
  sapphire_pickaxe:
    display_name: Sapphire Pickaxe
    resource:
      material: DIAMOND_PICKAXE
      generate: true
      model_id: 11
      textures:
      - item/sapphire_pickaxe.png
    attribute_modifiers:
      mainhand:
        attackDamage: 3.0
        attackSpeed: 1.2
  sapphire_hoe:
    display_name: Sapphire hoe
    resource:
      material: DIAMOND_HOE
      generate: true
      model_id: 7
      textures:
      - item/sapphire_hoe.png
    attribute_modifiers:
      mainhand:
        attackDamage: 1.0
        attackSpeed: 2.0
  sapphire_shovel:
    display_name: Sapphire Shovel
    resource:
      material: DIAMOND_SHOVEL
      generate: true
      model_id: 7
      textures:
      - item/sapphire_shovel.png
  emerald_axe:
    display_name: Emerald Axe
    resource:
      material: DIAMOND_AXE
      generate: true
      model_id: 8
      textures:
      - item/emerald_axe.png
    attribute_modifiers:
      mainhand:
        attackDamage: 8.0
        attackSpeed: 1.0
  emerald_pickaxe:
    display_name: Emerald Pickaxe
    resource:
      material: DIAMOND_PICKAXE
      generate: true
      model_id: 12
      textures:
      - item/emerald_pickaxe.png
    attribute_modifiers:
      mainhand:
        attackDamage: 3.0
        attackSpeed: 1.2
  emerald_hoe:
    display_name: Emerald hoe
    resource:
      material: DIAMOND_HOE
      generate: true
      model_id: 8
      textures:
      - item/emerald_hoe.png
    attribute_modifiers:
      mainhand:
        attackDamage: 1.0
        attackSpeed: 3.0
  emerald_shovel:
    display_name: Emerald Shovel
    resource:
      material: DIAMOND_SHOVEL
      generate: true
      model_id: 8
      textures:
      - item/emerald_shovel.png
  tinker_pickaxe:
    display_name: Tinker Pickaxe
    resource:
      material: DIAMOND_PICKAXE
      generate: false
      model_id: 16
      model_path: item/tinker_pickaxe
  tinker_axe:
    display_name: Tinker Axe
    resource:
      material: DIAMOND_PICKAXE
      generate: false
      model_id: 17
      model_path: item/tinker_axe
```

### Screenshots/Videos (you can drag and drop files or paste links)

https://user-images.githubusercontent.com/18208248/165909986-66419381-dc98-4e16-aa64-ae7705b1e0b6.mp4

**Comment by LoneDev6** [comment]
Will be fixed in 3.1.3
https://streamable.com/8j7vsc

**Comment by Bimic332** [comment]
the bug still exist
https://imgur.com/a/8b0Bjwb
